api/v1/validatorregistration_test.go

```diff
@@ -0,0 +1,168 @@
+		},
+
+		{
+			name:  "FeeRecipientMissing",
+			input: []byte(`{"gas_limit":"100","timestamp":"100","pubkey":"0x000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f202122232425262728292a2b2c2d2e2f"}`),
+			err:   "fee recipient missing",
+		},
+		{
+			name:  "FeeRecipientWrongType",
+			input: []byte(`{"fee_recipient":true,"gas_limit":"100","timestamp":"100","pubkey":"0x000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f202122232425262728292a2b2c2d2e2f"}`),
+			err:   "invalid JSON: json: cannot unmarshal bool into Go struct field validatorRegistrationJSON.fee_recipient of type string",
+		},
+		{
+			name:  "FeeRecipientInvalid",
+			input: []byte(`{"fee_recipient":"invalid","gas_limit":"100","timestamp":"100","pubkey":"0x000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f202122232425262728292a2b2c2d2e2f"}`),
+			err:   "invalid value for fee recipient: encoding/hex: invalid byte: U+0069 'i'",
+		},
+
+		{
+			name:  "GasLimitMissing",
+			input: []byte(`{"fee_recipient":"0x000102030405060708090a0b0c0d0e0f10111213","timestamp":"100","pubkey":"0x000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f202122232425262728292a2b2c2d2e2f"}`),
+			err:   "gas limit missing",
+		},
+		{
+			name:  "GasLimitWrongType",
+			input: []byte(`{"fee_recipient":"0x000102030405060708090a0b0c0d0e0f10111213","gas_limit":true,"timestamp":"100","pubkey":"0x000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f202122232425262728292a2b2c2d2e2f"}`),
+			err:   "invalid JSON: json: cannot unmarshal bool into Go struct field validatorRegistrationJSON.gas_limit of type string",
+		},
+		{
+			name:  "GasLimitInvalid",
+			input: []byte(`{"fee_recipient":"0x000102030405060708090a0b0c0d0e0f10111213","gas_limit":"-1","timestamp":"100","pubkey":"0x000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f202122232425262728292a2b2c2d2e2f"}`),
+			err:   "invalid value for gas limit: strconv.ParseUint: parsing \"-1\": invalid syntax",
+		},
+
+		{
+			name:  "TimestampMissing",
+			input: []byte(`{"fee_recipient":"0x000102030405060708090a0b0c0d0e0f10111213","gas_limit":"100","pubkey":"0x000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f202122232425262728292a2b2c2d2e2f"}`),
+			err:   "timestamp missing",
+		},
+		{
+			name:  "TimestampWrongType",
+			input: []byte(`{"fee_recipient":"0x000102030405060708090a0b0c0d0e0f10111213","gas_limit":"100","timestamp":true,"pubkey":"0x000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f202122232425262728292a2b2c2d2e2f"}`),
+			err:   "invalid JSON: json: cannot unmarshal bool into Go struct field validatorRegistrationJSON.timestamp of type string",
+		},
+		{
+			name:  "TimestampInvalid",
+			input: []byte(`{"fee_recipient":"0x000102030405060708090a0b0c0d0e0f10111213","gas_limit":"100","timestamp":"invalid","pubkey":"0x000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f202122232425262728292a2b2c2d2e2f"}`),
```

Comment:
time.unix() is of type int64 not sure how you want to test for invalid data - for now i've just used "invalid"